Add Aftermath mission In the Nick of Time

Smittytron opened the pull request:
Einstein needs to stop getting captured.

Smittytron committed:
Add Aftermath mission In the Nick of Time

darcyrush commented:
Mission plays well.

Minor suggestion; perhaps scientist and Einstein selection priority could be lowered

Smittytron commented:
> perhaps scientist and Einstein selection priority could be lowered

IMO that'd create an equally frustrating issue where players intend to grab the scientists and go but one or more combat units were in their dragbox.

tttppp commented:
Mission was good. At first it seemed excessively easy (jumping chronotanks into an empty base), but the scientist half of the mission was a bit harder.

On my first attempt I lost by destroying the forward command that the scientists come out of. I didn't spot anything in the briefing telling me not to blow this up and was confused as to why I lost. Maybe the failure message could be a bit more explicit: "Forward command containing scientists destroyed - mission failed."

The spy seems a bit pointless. I managed to disguise my spy and then immediately ran him onto a mine!

After the scientists are released then a Allied naval transport drops off some tanks, and these block one of the Soviet naval transports, sinking it before it can unload. I don't know if this is deliberate, but I would have expected to perform some quick thinking/micro to achieve this sort of thing rather than simply sitting and watching.

It's a bit weird losing control of my tanks at the extraction point. Is this deliberate?  I know the scientists are meant to leave the map like this, but I wanted to keep my tanks just in case I needed them.

tttppp commented on mods/ra/maps/in-the-nick-of-time/in-the-nick-of-time.lua:
```diff
@@ -0,0 +1,412 @@
+SouthWaterPath2 = { SouthWaterEntry.Location, SouthWaterLanding2.Location }
+SouthWaterTeam = { "dtrk", "1tnk", "1tnk", "2tnk", "2tnk" }
+SouthChinookPath = { ChinookEntry1.Location, ChinookLZ1.Location }
+SouthChinookChalk = { "e3", "e3", "e3", "e1", "e1" }
```
This took me a few attempts to get the reinforcements and was super disappointing.  I think it's better to just let the nuke truck blow it up.

Smittytron replied:
> The spy seems a bit pointless. I managed to disguise my spy and then immediately ran him onto a mine!

Every reinforcement is true to the original.

tttppp commented:
Lua code looks good to me.

Smittytron commented:
> On my first attempt I lost by destroying the forward command that the scientists come out of. I didn't spot anything in the briefing telling me not to blow this up and was confused as to why I lost. Maybe the failure message could be a bit more explicit: "Forward command containing scientists destroyed - mission failed."
> 
> After the scientists are released then a Allied naval transport drops off some tanks, and these block one of the Soviet naval transports, sinking it before it can unload. I don't know if this is deliberate, but I would have expected to perform some quick thinking/micro to achieve this sort of thing rather than simply sitting and watching.
> 
> It's a bit weird losing control of my tanks at the extraction point. Is this deliberate? I know the scientists are meant to leave the map like this, but I wanted to keep my tanks just in case I needed them.

Updated. I added a fail message explaining the scientists were in the command center. Changed the beach so the transport wouldn't be blocked. (Unless the player knows where the LZs are and intentionally blocks them. We can't fix that.) And I switched the ending to have only the scientists leave the map.

tttppp reviewed (dismissed):
I retested this, and everything is fixed. LGTM

Smittytron commented:
Rebased

tttppp reviewed (approved):
Retested and it still looks good!  Does this need retesting every rebase??

Smittytron commented:
> Retested and it still looks good! Does this need retesting every rebase??

Nah, a rebase with missions will always be the missions.yaml file. I just report 'Rebased' so that people know I didn't change anything related to the mission.

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f6ea7c76d272d5ecbbae1f7d9d76cdce47a5fbe0...29160b3df3171ba7b55b9687ddb986581c885371
